src/connection.ts

```diff
@@ -2821,7 +2821,14 @@ class Connection extends EventEmitter {
    */
   execBulkLoad(bulkLoad: BulkLoad) {
     bulkLoad.executionStarted = true;
+
+    const onCancel = () => {
+      request.cancel();
+    };
+
     const request = new Request(bulkLoad.getBulkInsertSql(), (error: (Error & { code?: string }) | null | undefined) => {
+      bulkLoad.removeListener('cancel', onCancel);
```

Comment:
This ensures we remove the event handler that forwarded cancellation from a `BulkLoad` object to the internal `Request` object.